draft-ietf-tls-cross-sni-resumption.md

```diff
@@ -116,6 +120,22 @@ merely lifts a performance-motivated "SHOULD NOT" recommendation from Section
 4.6.1.  Notably, it still relies on the client ensuring that the server
 certificate is valid for the new SNI at the time of session resumption.
 
+If the original server's assertion regarding supporting cross-name resumption
+turns out to be incorrect, the server receiving a misdirected ticket
+will not be able to decrypt it, and will therefore reject it.  This is secure,
+as session resumption may be safely rejected for any reason; however, such
+misconfiguration will waste tickets stored in the client's cache, as TLS
+tickets may be single-use.
+
+When providing the SNI value to the application, TLS 1.3 requires the value
+from the most recent ClientHello to be used ({{RFC8446, Section 4.6.1}}).  If the
+server TLS implementation violates that requirement and instead uses the legacy
+behavior of reporting the SNI value of the original session, this can lead to a
+confusion attack where the client and the server disagree on the server name
+being used (similar to the attacks described in {{DB15}}).  It is thus crucial
```

Comment:
"Legacy behaviour" here is something that you are assuming.  It is quite likely that many stacks would need to be updated to report the correct SNI (for both client and server), but you shouldn't label this "legacy".  Simply state that when an implementation supports this feature AND exercises it, then it needs to report the SNI that was used, not the one from the previous connection.  Failure to do so might lead to an attack (as noted).

Reply:
I think "legacy" here was meant to refer to the fact that certain TLS 1.2 implementations assumed SNI to be a session property, before 1.3 explicitly stated it was per-connection.  But you're right that "legacy" is ambiguous; reworded that.